ipn/ipnlocal/local_test.go

```diff
@@ -433,40 +428,3 @@ func (panicOnUseTransport) RoundTrip(*http.Request) (*http.Response, error) {
 }
 
 var nl = []byte("\n")
-
-func TestStartsInNeedsLoginState(t *testing.T) {
```

Comment:
Let's not delete tests to fix problems.

Reply:
This test exists only to confirm that something happens that absolutely should not happen. It has negative value.

Reply:
Well, it always started in NeedsLogin previously: https://github.com/tailscale/tailscale/issues/1759 as a side effect of hitting control and control saying "yeah, I don't know you"

So if we want to preserve the lazy machine key generation (which we do), then we need another way to start in NeedsLogin state to match what we did before.

So this test is still valid overall IMO.

Reply:
I'm pretty sure there's no way it "started" in the NeedsLogin state; I think it transitioned into the NeedsLogin state fairly quickly under certain conditions. The idea was that it shouldn't know whether it NeedsLogin until after it has first contacted the control server. If it was doing that in some recent version, that would explain the spurious "Needs Login" notifications on Windows when it didn't actually need login.

Anyway, I'm working on it. I thought it would be a quick fix, but I keep finding more stuff that could explain rare/oddball customer problems, so I'll keep looking deeper.